src/components/select/select.js

```diff
@@ -893,6 +912,12 @@ function OptgroupDirective() {
     compile: compile
   };
   function compile(el, attrs) {
+    // If we have a select header element, we don't want to add the normal label
```

Comment:
Can we refactor this function to separate out the concerns a bit and look more like the following?

```js
function compile(el, attrs) {
  if (!hasSelectHeader()) {
    setupLabelElement();
  }
}
```

I'm hesitant to use the `if (...) return` paradigm unless the condition truly excludes the _purpose_ of the function being run. The concern being that another developer may add something after the label setup not realizing that the header condition's `return` statement would let it be ignored.